src/valid/compiler/mod.rs

```diff
@@ -44,27 +43,23 @@ pub fn spec_compiler() -> Compiler {
 }
 
 /// Returns an error if v is not a valid path.
-fn is_path(v: &Value) -> Result<(), Box<dyn Error>> {
-    let Value::String(s) = v else {
-        Err("not a string")?
-    };
+fn is_path(v: &Value) -> Result<(), Box<dyn std::error::Error>> {
+    let Value::String(s) = v else { return Ok(()) };
 
     let path = RelativePath::new(s);
     for c in path.components() {
         if c == Component::ParentDir {
-            Err("parent dir")?
+            Err("references parent dir")?
         };
     }
 
     Ok(())
 }
 
-/// Returns an error if vi is not a valid SPDX license expression.
-fn is_license(v: &Value) -> Result<(), Box<dyn Error>> {
-    let Value::String(s) = v else {
-        Err("not a string")?
-    };
-    _ = spdx::Expression::parse(s)?;
+/// Returns an error if v is not a valid SPDX license expression.
+fn is_license(v: &Value) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
We'll keep `Box<dyn Error>` for these?

Reply:
Yes, these are JSONSchema format validation functions, which [is defined with this signature](https://github.com/santhosh-tekuri/boon/blob/main/src/formats.rs#L21):

```rust
    pub func: fn(v: &Value) -> Result<(), Box<dyn Error>>,
```

Reply:
Oooh okay

Perfect, then